src/stripeStreamingView.ts

```diff
@@ -0,0 +1,175 @@
+        this.stopStreaming();
+      }
+    }
+  };
+
+  stopStreaming = () => {
+    this.cleanupStreams();
+    this.setViewState(ViewState.Idle);
+  };
+
+  protected getStreamingControlItem(
+    viewName: string,
+    startCommand: string,
+    stopCommand: string,
+  ): StripeTreeItem {
+    const streamingControlItemArgs = (() => {
+      switch (this.viewState) {
+        case ViewState.Idle:
+          return {
+            label: `Start streaming ${viewName}`,
+            command: startCommand,
+            iconId: 'play-circle',
+          };
+        case ViewState.Loading:
+          return {
+            label: `Starting streaming ${viewName} ...`,
+            command: stopCommand,
+            iconId: 'loading',
+          };
+        case ViewState.Streaming:
+          return {
+            label: `Stop streaming ${viewName}`,
+            command: stopCommand,
+            iconId: 'stop-circle',
+          };
+      }
+    })();
+
+    return this.createItemWithCommand(streamingControlItemArgs);
+  }
+
+  private createItemWithCommand({
+    label,
+    command,
+    iconId,
+  }: {
+    label: string;
+    command?: string;
+    iconId?: string;
+  }) {
+    const item = new StripeTreeItem(label, {
+      commandString: command,
+      iconPath: iconId ? new ThemeIcon(iconId) : undefined,
+    });
+    return item;
+  }
+
+  private async setupStreams() {
+    const tailProcess = await this.createStreamProcess();
+    tailProcess.on('exit', this.stopStreaming);
+
+    if (!this.stderrStream) {
+      this.stderrStream = new stream.Writable({
+        write: (chunk, _, callback) => {
+          if (this.streamReady(chunk)) {
+            this.setViewState(ViewState.Streaming);
+          } else if (!this.streamLoading(chunk)) {
+            window.showErrorMessage(chunk);
+            this.stopStreaming();
+          }
+          callback();
+        },
+        decodeStrings: false,
+      });
+      tailProcess.stderr.setEncoding('utf8').pipe(new LineStream()).pipe(this.stderrStream);
+    }
+
+    if (!this.stdoutStream) {
+      this.stdoutStream = new stream.Writable({
+        write: (chunk, _, callback) => {
+          try {
+            const object = this.createTreeItem(chunk);
+            if (object) {
+              this.insertItem(object);
+            }
+          } catch {}
+          callback();
+        },
+        decodeStrings: false,
+      });
+      tailProcess.stdout.setEncoding('utf8').pipe(new LineStream()).pipe(this.stdoutStream);
+    }
+  }
+
+  // Tell us how to start the process
+  abstract createStreamProcess(): Promise<ChildProcess>;
+
+  // Tell us how we can tell when the process is ready
+  abstract streamReady(chunk: any): boolean;
+
+  // Tell us how we can tell if the stream is still loading
+  abstract streamLoading(chunk: any): boolean;
+
+  // Tell us how to process the chunk into a tree item.
+  abstract createTreeItem(chunk: any): StripeTreeItem | null;
```

Comment:
Can we make this name more specific? For example, `createStreamTreeItem` or something to that effect.

Reply:
absolutely! Will update